docs/user/trusted-publishers/internals.md

```diff
@@ -176,8 +179,31 @@ used by the Python packaging community:
    can be published from both `release-linux.yml` and `release-macos.yml`
    without needing to be refactored into a single `release.yml`.
 
+### What are account resurrection attacks, and how does PyPI protect against them?
+
+Some OIDC providers support username changes, so a claim of
+`repository_owner: octo-org` might not necessarily refer to the same `octo-org`
+that a user initially authorized in a trusted publisher configuration.
+
+If a repository owner changes their username or deletes their account, a
+malicious actor can take the freed username and create their own repositories
+under the original trusted name. This is known as an *account resurrection
+attack*.
```

Comment:
Clarify that it's a possibility (not all IdPs allow username reuse, or may forbid it in certain circumstances):

```suggestion
If a repository owner changes their username or deletes their account, a
malicious actor may be able to take the freed username and create their own repositories
under the original trusted name. This is known as an *account resurrection
attack*.
```